02-peer-protocol.md

```diff
@@ -2470,14 +2994,18 @@ A node:
 
 A receiving node:
   - if `next_funding_txid` is set:
-    - if `next_funding_txid` matches the latest interactive funding transaction:
+    - if `next_funding_txid` matches the latest interactive funding transaction
+      or the current channel funding transaction:
       - if it has not received `tx_signatures` for that funding transaction:
         - MUST retransmit its `commitment_signed` for that funding transaction.
         - if it has already received `commitment_signed` and it should sign first,
           as specified in the [`tx_signatures` requirements](#the-tx_signatures-message):
           - MUST send its `tx_signatures` for that funding transaction.
       - if it has already received `tx_signatures` for that funding transaction:
         - MUST send its `tx_signatures` for that funding transaction.
+    - if `next_funding_txid` matches the latest funding transaction:
+      - if that transaction has reached acceptable depth:
+        - MUST send `splice_locked`.
     - otherwise:
       - MUST send `tx_abort` to let the sending node know that they can forget
         this funding transaction.
```

Comment:
We're missing the case where an invalid `next_funding_txid` is sent and we should fail the channel. Here is my attempt to add it in. The conditions are very difficult to articulate here so any ideas for rewording are welcome.

```
A receiving node:
  - if `next_funding_txid` is set:
    - if `next_funding_txid` matches the latest interactive funding transaction:
      - if it has not received `tx_signatures` for that funding transaction:
        - MUST retransmit its `commitment_signed` for that funding transaction.
        - if it has already received `commitment_signed` and it should sign first,
          as specified in the [`tx_signatures` requirements](#the-tx_signatures-message):
          - MUST send its `tx_signatures` for that funding transaction.
      - if it has already received `tx_signatures` for that funding transaction:
        - MUST send its `tx_signatures` for that funding transaction.
    - otherwise:
      - if there is no pending local interactive funding transaction:
        - if `next_funding_txid` matches the latest funding transaction:
          - MUST send `splice_locked`.
        - otherwise:
          - MUST send `tx_abort` to let the sending node know that they can forget
        this funding transaction.
      - otherwise:
        - MUST fail the channel
```

Without some clause like this the channel can get into an infinite disconnect / reconnect cycle which will mess up any other channels we have with the same peer.

Reply:
Good catch! Indeed if we reconnect and both sides are asking for a `next_funding_txid`, but it's not the same, then one side is seriously wrong and we should force-close. I'll update that.

Reply:
Done in https://github.com/lightning/bolts/pull/1160/commits/aece0d8f42310d7e7d128905cf18eb0414c02459